In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('7817_1.csv')

In [3]:
df

id       asins   brand  \
0     AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon   
1     AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon   
2     AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon   
3     AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon   
4     AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon   
...                    ...         ...     ...   
1592  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1593  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1594  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1595  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1596  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   

                                             categories colors  \
0                            Amazon Devices,mazon.co.uk    NaN   
1                            Amazon Devices,mazon.co.uk    NaN   
2                            Amazon Devices,mazon.co.uk    NaN   
3                            Amazon Devices,mazon.co.uk    NaN   
4                            Amazon Devices,mazon.co.uk    NaN   
...                                                 ...    ...   
1592  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1593  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1594  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1595  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1596  Amazon Devices & Accessories,Amazon Device Acc...    NaN   

                 dateAdded           dateUpdated                 dimension  \
0     2016-03-08T20:21:53Z  2017-07-18T23:52:58Z  169 mm x 117 mm x 9.1 mm   
1     2016-03-08T20:21:53Z  2017-07-18T23:52:58Z  169 mm x 117 mm x 9.1 mm   
2     2016-03-08T20:21:53Z  2017-07-18T23:52:58Z  169 mm x 117 mm x 9.1 mm   
3     2016-03-08T20:21:53Z  2017-07-18T23:52:58Z  169 mm x 117 mm x 9.1 mm   
4     2016-03-08T20:21:53Z  2017-07-18T23:52:58Z  169 mm x 117 mm x 9.1 mm   
...                    ...                   ...                       ...   
1592  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z                       NaN   
1593  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z                       NaN   
1594  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z                       NaN   
1595  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z                       NaN   
1596  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z                       NaN   

      ean                                               keys  ...  \
0     NaN                        kindlepaperwhite/b00qjdu3ky  ...   
1     NaN                        kindlepaperwhite/b00qjdu3ky  ...   
2     NaN                        kindlepaperwhite/b00qjdu3ky  ...   
3     NaN                        kindlepaperwhite/b00qjdu3ky  ...   
4     NaN                        kindlepaperwhite/b00qjdu3ky  ...   
...   ...                                                ...  ...   
1592  NaN  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...   
1593  NaN  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...   
1594  NaN  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...   
1595  NaN  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...   
1596  NaN  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...   

     reviews.rating                                 reviews.sourceURLs  \
0               5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
1               5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
2               4.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
3               5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
4               5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
...             ...                                                ...   
1592            3.0  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1593            1.0  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1594            1.0  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1595            3.0  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1596            1.0  https://www.amazon.com/Alexa-

In [4]:
df.head()

id       asins   brand                  categories  \
0  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
1  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
2  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
3  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
4  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   

  colors             dateAdded           dateUpdated  \
0    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
1    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
2    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
3    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
4    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   

                  dimension  ean                         keys  ...  \
0  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
1  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
2  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
3  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
4  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   

  reviews.rating                                 reviews.sourceURLs  \
0            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
1            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
2            4.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
3            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
4            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   

                                        reviews.text  \
0  I initially had trouble deciding between the p...   
1  Allow me to preface this with a little history...   
2  I am enjoying it so far. Great for reading. Ha...   
3  I bought one of the first Paperwhites and have...   
4  I have to say upfront - I don't like coroporat...   

                                reviews.title reviews.userCity  \
0              Paperwhite voyage, no regrets!              NaN   
1           One Simply Could Not Ask For More              NaN   
2  Great for those that just want an e-reader              NaN   
3                    Love / Hate relationship              NaN   
4                                   I LOVE IT              NaN   

  reviews.userProvince    reviews.username  sizes upc     weight  
0                  NaN          Cristina M    NaN NaN  205 grams  
1                  NaN               Ricky    NaN NaN  205 grams  
2                  NaN       Tedd Gardiner    NaN NaN  205 grams  
3                  NaN              Dougal    NaN NaN  205 grams  
4                  NaN  Miljan David Tanic    NaN NaN  205 grams  

[5 rows x 27 columns]

In [5]:
df.tail()

id       asins   brand  \
1592  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1593  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1594  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1595  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   
1596  AVpfo9ukilAPnD_xfhuj  B00NO8JJZW  Amazon   

                                             categories colors  \
1592  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1593  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1594  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1595  Amazon Devices & Accessories,Amazon Device Acc...    NaN   
1596  Amazon Devices & Accessories,Amazon Device Acc...    NaN   

                 dateAdded           dateUpdated dimension  ean  \
1592  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z       NaN  NaN   
1593  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z       NaN  NaN   
1594  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z       NaN  NaN   
1595  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z       NaN  NaN   
1596  2016-04-02T14:40:43Z  2017-08-13T08:28:46Z       NaN  NaN   

                                                   keys  ... reviews.rating  \
1592  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...            3.0   
1593  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...            1.0   
1594  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...            1.0   
1595  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...            3.0   
1596  alexavoiceremoteforamazonfiretvfiretvstick/b00...  ...            1.0   

                                     reviews.sourceURLs  \
1592  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1593  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1594  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1595  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   
1596  https://www.amazon.com/Alexa-Voice-Remote-Amaz...   

                                           reviews.text  \
1592  This is not the same remote that I got for my ...   
1593  I have had to change the batteries in this rem...   
1594  Remote did not activate, nor did it connect to...   
1595  It does the job but is super over priced. I fe...   
1596  I ordered this item to replace the one that no...   

                                          reviews.title reviews.userCity  \
1592  I would be disappointed with myself if i produ...              NaN   
1593                        Battery draining remote!!!!              NaN   
1594      replacing an even worse remote. Waste of time              NaN   
1595                                         Overpriced              NaN   
1596  I am sending all of this crap back to amazon a...              NaN   

     reviews.userProvince  reviews.username  sizes upc    weight  
1592                  NaN   GregAmandawith4    NaN NaN  4 ounces  
1593                  NaN   Amazon Customer    NaN NaN  4 ounces  
1594                  NaN   Amazon Customer    NaN NaN  4 ounces  
1595                  NaN        Meg Ashley    NaN NaN  4 ounces  
1596                  NaN           DIANE K    NaN NaN  4 ounces  

[5 rows x 27 columns]

In [6]:
df.isnull()

id  asins  brand  categories  colors  dateAdded  dateUpdated  \
0     False  False  False       False    True      False        False   
1     False  False  False       False    True      False        False   
2     False  False  False       False    True      False        False   
3     False  False  False       False    True      False        False   
4     False  False  False       False    True      False        False   
...     ...    ...    ...         ...     ...        ...          ...   
1592  False  False  False       False    True      False        False   
1593  False  False  False       False    True      False        False   
1594  False  False  False       False    True      False        False   
1595  False  False  False       False    True      False        False   
1596  False  False  False       False    True      False        False   

      dimension   ean   keys  ...  reviews.rating  reviews.sourceURLs  \
0         False  True  False  ...           False               False   
1         False  True  False  ...           False               False   
2         False  True  False  ...           False               False   
3         False  True  False  ...           False               False   
4         False  True  False  ...           False               False   
...         ...   ...    ...  ...             ...                 ...   
1592       True  True  False  ...           False               False   
1593       True  True  False  ...           False               False   
1594       True  True  False  ...           False               False   
1595       True  True  False  ...           False               False   
1596       True  True  False  ...           False               False   

      reviews.text  reviews.title  reviews.userCity  reviews.userProvince  \
0            False          False              True                  True   
1            False          False              True                  True   
2            False          False              True                  True   
3            False          False              True                  True   
4            False          False              True                  True   
...            ...            ...               ...                   ...   
1592         False          False              True                  True   
1593         False          False              True                  True   
1594         False          False              True                  True   
1595         False          False              True                  True   
1596         False          False              True                  True   

      reviews.username  sizes   upc  weight  
0                False   True  True   False  
1                False   True  True   False  
2                False   True  True   False  
3                False   True  True   False  
4                False   True  True   False  
...                ...    ...   ...     ...  
1592             False   True  True   False  
1593             False   True  True   False  
1594             False   True  True   False  
1595             False   True  True   False  
1596             False   True  True   False  

[1597 rows x 27 columns]

In [12]:
df.isnull().sum()

id                         0
asins                      0
brand                      0
categories                 0
colors                   823
dateAdded                  0
dateUpdated                0
dimension               1032
ean                      699
keys                       0
manufacturer             632
manufacturerNumber       695
name                       0
prices                     0
reviews.date             380
reviews.doRecommend     1058
reviews.numHelpful       697
reviews.rating           420
reviews.sourceURLs         0
reviews.text               0
reviews.title             17
reviews.userCity        1597
reviews.userProvince    1597
reviews.username          17
sizes                   1597
upc                      699
weight                   911
dtype: int64

In [13]:
df.isnull().sum().sum()

12871

# We have total 12871 null values in given data set

In [14]:
df.describe()

ean  reviews.numHelpful  reviews.rating  reviews.userCity  \
count  8.980000e+02          900.000000     1177.000000               0.0   
mean   8.443135e+11           83.584444        4.359388               NaN   
std    3.416444e+09          197.150238        1.021445               NaN   
min    8.416670e+11            0.000000        1.000000               NaN   
25%    8.416670e+11            0.000000        4.000000               NaN   
50%    8.416670e+11            0.000000        5.000000               NaN   
75%    8.487190e+11           34.000000        5.000000               NaN   
max    8.487190e+11          997.000000        5.000000               NaN   

       reviews.userProvince  sizes           upc  
count                   0.0    0.0  8.980000e+02  
mean                    NaN    NaN  8.443135e+11  
std                     NaN    NaN  3.416444e+09  
min                     NaN    NaN  8.416670e+11  
25%                     NaN    NaN  8.416670e+11  
50%                     NaN    NaN  8.416670e+11  
75%                     NaN    NaN  8.487190e+11  
max                     NaN    NaN  8.487190e+11

In [16]:
df.shape

(1597, 27)

# In this dataset we have 1597 rows and 27 columns

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1597 entries, 0 to 1596
Data columns (total 27 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    1597 non-null   object 
 1   asins                 1597 non-null   object 
 2   brand                 1597 non-null   object 
 3   categories            1597 non-null   object 
 4   colors                774 non-null    object 
 5   dateAdded             1597 non-null   object 
 6   dateUpdated           1597 non-null   object 
 7   dimension             565 non-null    object 
 8   ean                   898 non-null    float64
 9   keys                  1597 non-null   object 
 10  manufacturer          965 non-null    object 
 11  manufacturerNumber    902 non-null    object 
 12  name                  1597 non-null   object 
 13  prices                1597 non-null   object 
 14  reviews.date          1217 non-null   object 
 15  reviews.doRecommend  

In [21]:
df[df.duplicated()].shape

(0, 27)

# We have 27 duplicated rows

In [31]:
list(df.id.unique())

['AVpe7AsMilAPnD_xQ78G',
 'AVpfBEWcilAPnD_xTGb7',
 'AVpff7_VilAPnD_xc1E_',
 'AV00lzP7GV-KLJ3ac0uk',
 'AV1T0_7M-jtxr-f31CoX',
 'AVpfpzCi1cnluZ0-oxEr',
 'AV00l7jV-jtxr-f30lnX',
 'AV1T09XZ-jtxr-f31CoT',
 'AV3as-VX-jtxr-f3_PPu',
 'AV00lzd5GV-KLJ3ac0ul',
 'AV1T1nWrglJLPUi8H8Yz',
 'AV1T09fyvKc47QAVgf2R',
 'AV1T0-J7vKc47QAVgf2T',
 'AV1T0_ti-jtxr-f31CoV',
 'AVph5AshLJeJML43f_fT',
 'AV1T0_tnvKc47QAVgf2V',
 'AV1T19jBvKc47QAVgf3S',
 'AVpg4dIu1cnluZ0-7fWM',
 'AV2Mus6LGV-KLJ3alnyc',
 'AV3atCk2glJLPUi8SQna',
 'AVphEeRyilAPnD_x035P',
 'AV29VS9cvKc47QAVp03c',
 'AV000tWuGV-KLJ3ac2-b',
 'AV1T1nR5-jtxr-f31CpL',
 'AV1T19k3GV-KLJ3adSMS',
 'AV1T19lYglJLPUi8H8Y_',
 'AV29ePzlGV-KLJ3amjUT',
 'AV1T19jOglJLPUi8H8Y9',
 'AV1T19mDGV-KLJ3adSMT',
 'AV1T1u7jvKc47QAVgf3J',
 'AV1T19kvglJLPUi8H8Y-',
 'AV1Nik13-jtxr-f31AFO',
 'AVpfldWhLJeJML43_7zF',
 'AV1T1nJ_GV-KLJ3adSMC',
 'AVpfLiCSilAPnD_xWpk_',
 'AVsRjfwAU2_QcyX9PHqe',
 'AVzRloqLGV-KLJ3aavBd',
 'AVpe6TZCilAPnD_xQrVy',
 'AVzRkFTFvKc47QAVd43-',
 'AV1T1nWFGV-KLJ3adSMD',
